src/geowombat/core/api.py

```diff
@@ -511,6 +526,19 @@ def __init__(
 
                     with xr.open_dataset(filename, **kwargs) as src:
                         self.data = src.to_array(dim='band')
+                    if band_names is not None:

```

Comment:
This block allows the user to pass a desired band order after reading data from a NetCDF dataset (which might reorder the variables by name).